main.js

```diff
@@ -124,3 +128,27 @@ function getView(views, id) {
     }
   }
 }
+
+function addToCartButton(product, itemList) {
+  var addToCartButton = document.createElement('button')
+  addToCartButton.setAttribute('type', 'button')
+  addToCartButton.setAttribute('id', 'add-to-cart')
+  addToCartButton.classList.add('btn')
+  addToCartButton.classList.add('btn-danger')
+  addToCartButton.textContent = 'Add to Cart'
+  product.appendChild(addToCartButton)
+  addToCartButton.addEventListener('click', function(event){
+    addToCart(product, shoppingCart, itemList)
+    updateCartButton(shoppingCart)
+  })
+}
+
+function addToCart(product, cart, itemList) {
+  var cartId = product.id
+  cart.unshift(itemList[cartId]);
+}
+
+function updateCartButton(cartSize) {
+  count = cartSize.length
```

Comment:
don't forget to declare `var count`, otherwise you're declaring a global variable